homeassistant/components/modbus/binary_sensor.py

```diff
@@ -61,10 +78,41 @@ async def async_update(self, now: datetime | None = None) -> None:
                 return
             self._lazy_errors = self._lazy_error_count
             self._attr_available = False
-            self.async_write_ha_state()
-            return
+            result_bits = None
+        else:
+            self._lazy_errors = self._lazy_error_count
+            self._attr_is_on = result.bits[0] & 1
+            self._attr_available = True
+            result_bits = result.bits
 
-        self._lazy_errors = self._lazy_error_count
-        self._attr_is_on = result.bits[0] & 1
-        self._attr_available = True
+        self.schedule_update_ha_state()
```

Comment:
You're in async context, you can use async function to write to state right away. 
```suggestion
        self.async_write_ha_state()
```

Reply:
Actually, you're already doing that in the next line. Why would you want to schedule an update of the HA state? 
```suggestion
```

Reply:
ups, that's a leftover, removed.